api/v1/mongodbcommunity_types.go

```diff
@@ -69,7 +69,7 @@ type MongoDBCommunitySpec struct {
 	Users []MongoDBUser `json:"users"`
 
 	// +optional
-	StatefulSetConfiguration StatefulSetConfiguration `json:"statefulSet,omitempty"`
+	StatefulSetConfiguration appsv1.StatefulSetSpec `json:"statefulSet,omitempty"`
```

Comment:
This change fixed the statefulset_arbitrary e2e tests. For some reason, even though the custom marshal/unmarshal  code should be including this field when annotated with "-", it is being ignored. For that reason I reintroduced the CRD generation (which should be fine as long as we can automatically generate the CRD).

Reply:
we need to make sure the end resource still works by specifying `statefulSet.spec`